go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -708,3 +719,13 @@ func (te *TxEngine) beginNewDbaConnection(ctx context.Context) (*StatefulConnect
 	_, _, err = te.txPool.begin(ctx, nil, false, sc, nil)
 	return sc, err
 }
+
+// IsTwoPCAllowed checks if TwoPC is allowed.
+func (te *TxEngine) IsTwoPCAllowed() bool {
+	for _, allowed := range te.twopcAllowed {
+		if !allowed {
+			return false
+		}
+	}
```

Comment:
OK, so the different ones don't matter here. It's true/false. Am I missing where we benefit from the new slice?

Reply:
The two can be toggled independently. Like if the reason for two pc being disabled is that semi-sync is turned off. The code that is refreshing from topo shouldn't turn it on. Essentially they are 2 separate states. We could store 2 boolean values separately like `twoPcAllowedSemiSycn` and `twoPcAllowsTabletControls` but we thought a single array is better in the sense that it is a single field and less likely to be missed if we create/augment a function.